src/components/common/SystemCommands.vue

```diff
@@ -161,9 +161,19 @@ export default class SystemCommands extends Mixins(StateMixin, ServicesMixin) {
       })
   }
 
-  togglePowerDevice (device: Device, wait?: string) {
-    const state = (device.status === 'on') ? 'off' : 'on'
-    SocketActions.machineDevicePowerToggle(device.device, state, wait)
+  async togglePowerDevice (device: Device, wait?: string) {
+    const confirmOnPowerDeviceChange = this.$store.state.config.uiSettings.general.confirmOnPowerDeviceChange
+    let res: boolean | undefined = true
+    if (confirmOnPowerDeviceChange) {
+      res = await this.$confirm(
+        this.$tc('app.general.simple_form.msg.confirm_power_device_toggle'),
+        { title: this.$tc('app.general.label.confirm'), color: 'card-heading', icon: '$error' }
+      )
+    }
```

Comment:
Really not a fan of this tbh, maybe something like

```suggestion
    if (
      confirmOnPowerDeviceChange
      && !await this.$confirm(
        this.$tc('app.general.simple_form.msg.confirm_power_device_toggle'),
        {
          title: this.$tc('app.general.label.confirm'),
          color: 'card-heading',
          icon: '$error'
      	}
      )
    ) {
      return
    }
```

Reply:
I personally prefer writing as you did above, but the reason I wrote it as I did was for consistency purposes, as there was already code written that way!

Reply:
i agree with you both.  however, let's keep it consistent with existing coding techniques for now...